java/src/jni/h5util.c

```diff
@@ -2910,6 +3054,124 @@ h5str_dump_simple_dset(JNIEnv *env, FILE *stream, hid_t dset, int binary_order)
     return ret_value;
 } /* end h5str_dump_simple_dset */
 
+int
+h5str_dump_simple_mem(JNIEnv *env, FILE *stream, hid_t attr_id, int binary_order)
+{
+    hid_t          f_space = H5I_INVALID_HID; /* file data space */
+    hsize_t        alloc_size;
+    int            ndims;                    /* rank of dataspace */
+    unsigned       i;                        /* counters  */
+    hsize_t        total_size[H5S_MAX_RANK]; /* total size of dataset*/
+    hsize_t        p_nelmts;                 /* total selected elmts */
+    unsigned char *buf = NULL;               /* buffer for raw data */
+
+    /* VL data special information */
+    unsigned int vl_data = 0; /* contains VL datatypes */
+    hid_t        p_type  = H5I_INVALID_HID;
+    hid_t        f_type  = H5I_INVALID_HID;
+
+    int ret_value = FAIL;
+
+    if (attr_id < 0)
+        H5_BAD_ARGUMENT_ERROR(ENVONLY, "h5str_dump_simple_mem: attr ID < 0");
+
+    if ((f_type = H5Aget_type(attr_id)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    switch (binary_order) {
+        case 1: {
+            if ((p_type = h5str_get_native_type(f_type)) < 0)
+                CHECK_JNI_EXCEPTION(ENVONLY, JNI_FALSE);
+
+            break;
+        }
+
+        case 2: {
+            if ((p_type = h5str_get_little_endian_type(f_type)) < 0)
+                CHECK_JNI_EXCEPTION(ENVONLY, JNI_FALSE);
+
+            break;
+        }
+
+        case 3: {
+            if ((p_type = h5str_get_big_endian_type(f_type)) < 0)
+                CHECK_JNI_EXCEPTION(ENVONLY, JNI_FALSE);
+
+            break;
+        }
+
+        default: {
+            if ((p_type = H5Tcopy(f_type)) < 0)
+                H5_LIBRARY_ERROR(ENVONLY);
+
+            break;
+        }
+    }
+
+    if ((f_space = H5Aget_space(attr_id)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    if ((ndims = H5Sget_simple_extent_ndims(f_space)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    if (H5Sget_simple_extent_dims(f_space, total_size, NULL) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    /* Calculate the number of elements we're going to print */
+    p_nelmts = 1;
+
+    if (ndims > 0) {
+        for (i = 0; i < (size_t)ndims; i++)
+            p_nelmts *= total_size[i];
+    } /* end if */
+
+    if (p_nelmts > 0) {
+        /* Check if we have VL data in the dataset's datatype */
+        if (h5str_detect_vlen(p_type) != 0)
+            vl_data = 1;
+
+        alloc_size = p_nelmts * H5Tget_size(p_type);
+        if (alloc_size > 0) {
+            if (NULL != (buf = (unsigned char *)HDmalloc((size_t)alloc_size))) {
+                H5_OUT_OF_MEMORY_ERROR(ENVONLY, "h5str_dump_simple_mem: failed to allocate sm_buf");
```

Comment:
Is this routine used anywhere? The above line is always going to throw an "out of memory error".

Reply:
Part of export_attribute - will remove.